activerecord/lib/active_record/connection_adapters/sql_type_metadata.rb

```diff
@@ -29,6 +29,12 @@ def hash
         def attributes_for_hash
           [self.class, sql_type, type, limit, precision, scale]
         end
+
+      private
+
+        def intern(value)
+          value.is_a?(String) ? -value : value
```

Comment:
Aren't these only called on String or nil? Would it be clearer with `value.nil? ? -value : value`?

Reply:
Most of the time yes, but I had cases where the `type` was a Symbol in the test suite. I suspect the test is actually wrong, but I'm not sure how to confirm.

Reply:
I think it is better to keep the pattern the same so we can possibly extract later.

Reply:
Which test was it that failed?

Reply:
https://buildkite.com/rails/rails/builds/60127#2b732ada-6845-46a0-8f52-fcf34244f6f4/119-149

Caused by this IIRC: https://github.com/rails/rails/blob/dd972f9efe832e882c559ebc92274e3e5ccc294d/activerecord/test/models/contact.rb#L13-L20

Reply:
> so we can possibly extract later.

I think ActiveSupport could generalize the `-@` interface. especially for immutable types.

Having `NilClass#-@` and `Symbol#-@` would make it so that I don't really have to worry about the type.

Reply:
@rafaelfranca yes, I was implying that every one of the intern methods maybe could be written that way.

Reply:
@casperisfine what would `Symbol#-@` return? The general case I'm worrying about is where people pass in a symbol one time and a string the next, but that's probably not a big deal.

Reply:
@casperisfine I can't find the `column` method for `ActiveRecord::Base` objects on api.rubyonrails.org, so I take it it's private. We should find the `source_location` and double check though.

Reply:
> what would Symbol#-@ return? 

Itself.

`String#-@` if you ignore the deduplication (because symbols don't need it) is:

```ruby
def -@
  frozen? ? self : freeze
end
```

So my suggestion would be:

```
def -@
  self
end
```

for all immutable types: `NilClass / TrueClass / FalseClass / Symbols / Integer`.

This way if you know the value is a simple value and not an array, hash or whatver, you can blindly call `-@`.

Reply:
@kaspth it's defined it that model itself: https://github.com/rails/rails/blob/dd972f9efe832e882c559ebc92274e3e5ccc294d/activerecord/test/models/contact.rb#L28-L31

it's a pure test method. Just a shortcut for `connection.merge_column`

Reply:
> it's a pure test method. Just a shortcut for `connection.merge_column`

Jebus, I'm getting tired! 😴